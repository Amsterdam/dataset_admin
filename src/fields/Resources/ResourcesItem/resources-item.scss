@import "../../../styling/config.scss";

.resources-item {
  @extend %base-font;
  flex-grow: 1;
  position: relative;

  &__title {
    @extend %heavy-font;
    color: $brownish-grey;
    margin-bottom: 2px;
  }

  &__description {
    color: $black;
    min-height: 22px;

    &-text {
      overflow: hidden;
      position: absolute;
      text-overflow: ellipsis;
      white-space: nowrap;
      width: calc(100% - 200px);
    }
  }

  &__modified {
    color: $warm-grey;
    font-size: 14px;
    line-height: 23px;
    margin-bottom: 2px;
    position: absolute;
    right: 39px;
    top: 0;

    &::before {
      content: "ververst op ";
    }

    &:empty {
      &::before {
        display: none;
      }
    }
  }

  &__file-size {
    bottom: 0;
    color: $black;
    position: absolute;
    right: 60px;

    &::after {
      @extend %icon-arrow-right;
      bottom: -5px;
      content: "";
      display: inline-block;
      height: 35px;
      position: absolute;
      width: 35px;
    }
  }

  &__file-type {
    background-color: $pinkish-grey;
    color: $black;
    display: inline-block;
    font-size: 14px;
    line-height: 1;
    margin-right: 5px;
    padding: 2px 5px 0 5px;

    &--anders {
      background-color: $brownish-grey;
      color: $white;
    }

    &--csv {
      background-color: $aquamarine;
    }

    &--docx {
      background-color: $darkish-blue;
      color: $white;
    }

    &--geojson {
      background-color: $marigold;
    }

    &--gml {
      background-color: $deep-green-two;
      color: $white;
    }

    &--html {
      background-color: $azure;
    }

    &--json {
      background-color: $dandelion;
    }

    &--pdf {
      background-color: $red;
      color: $white;
    }

    &--png {
      background-color: $turquoise-blue;
    }

    &--rest-atom-feed {
      background-color: $tangerine;
    }

    &--shp {
      background-color: $deep-green;
      color: $white;
    }

    &--website {
      background-color: $warm-grey;
    }

    &--wfs {
      background-color: $red-violet;
      color: $white;
    }

    &--wms {
      background-color: $dark-hot-pink;
      color: $white;
    }

    &--xml {
      background-color: $ugly-yellow;
    }

    &--xls,
    &--xlsx {
      background-color: $emerald;
    }
  }
}
