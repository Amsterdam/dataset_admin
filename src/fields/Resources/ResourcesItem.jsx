import React from 'react';
import PropTypes from 'prop-types';

import filesize from 'filesize';
import { dateFormat } from '../../definitions/localization';

import './resources-item.scss';

function getFileType(mime, fieldSchema) {
  const type = fieldSchema.enumNames[fieldSchema.enum.indexOf(mime)];
  return (
    <span className={`resources-item__file-type
      resources-item__file-type--${type}`}
    >{type || 'overig'}
    </span>
  );
}

const ResourcesItem = ({ resource, schemaProps }) => (
  <div className="resources-item">
    <div className="resources-item__modified">
      {dateFormat.formatDate(resource['dct:modified'])}</div>
    <div className="resources-item__file-size">
      {resource['dcat:byteSize'] > 0 ? filesize(resource['dcat:byteSize']) : ''}</div>
    <div className="resources-item__title">
      {resource['dct:title']}</div>
    <div className="resources-item__description">
      {getFileType(resource['dct:format'], schemaProps['dct:format'])}
      {resource['dct:description']}</div>
  </div>
);

ResourcesItem.defaultProps = {
  resource: {}
};

ResourcesItem.propTypes = {
  resource: PropTypes.object,
  schemaProps: PropTypes.object.isRequired
};

export default ResourcesItem;
