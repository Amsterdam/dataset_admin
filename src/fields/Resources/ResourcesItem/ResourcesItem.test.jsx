import React from 'react';
import { shallow } from 'enzyme';

import ResourcesItem from './ResourcesItem';

const schemaProps = {
  'dct:format': {
    enum: [
      'n/a',
      'text/csv',
      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
      'application/vnd.geo+json',
      'application/gml+xml',
      'text/html',
      'application/json',
      'application/pdf',
      'image/png',
      'application/vnd.ms-excel',
      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
      'application/xml',
      'application/octet-stream'
    ],
    enumNames: ['', 'CSV', 'DOCX', 'GeoJSON', 'GML', 'HTML', 'JSON', 'PDF', 'PNG', 'XLS', 'XLSX', 'XML', 'Anders']
  }
};

/*
resource = {
  'dct:title': 'Titel',
  'dcat:accessURL': 'http://ergens',
  'dct:description': 'omschrijving',
  'dct:format': 'application/pdf',
  'dcat:byteSize': '666',
  'foaf:isPrimaryTopicOf': {
    'dct:modified': '2017-11-30'
  }
  */

describe('The ResourcesItem component', () => {
  it('renders with title, description, format, file size and date', () => {
    const resource = {
      'dct:title': 'Titel',
      'dcat:accessURL': 'http://ergens',
      'dct:description': 'omschrijving',
      'dct:format': 'application/pdf',
      'dcat:byteSize': '666',
      'foaf:isPrimaryTopicOf': {
        'dct:modified': '2017-11-30'
      }
    };

    const wrapper = shallow(
      <ResourcesItem
        resource={resource}
        schemaProps={schemaProps}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with title and url: showing url in place of description', () => {
    const resource = {
      'dct:title': 'Titel',
      'dcat:accessURL': 'http://ergens',
      'dct:format': 'n/a',
      'foaf:isPrimaryTopicOf': {}
    };

    const wrapper = shallow(
      <ResourcesItem
        resource={resource}
        schemaProps={schemaProps}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
