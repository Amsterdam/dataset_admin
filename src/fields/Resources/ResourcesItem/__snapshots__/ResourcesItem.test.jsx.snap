// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResourcesItem component renders with title and url: showing url in place of description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "ams:purl": "http://acc.ergens",
        "dcat:accessURL": "http://ergens",
        "dcat:mediaType": "n/a",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {},
      }
    }
    schemaProps={
      Object {
        "ams:distributionType": Object {
          "enum": Array [
            "api",
            "file",
            "web",
          ],
          "enumNames": Array [
            "API/Service",
            "Bestand",
            "Website",
          ],
        },
        "dcat:mediaType": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__file-type resources-item__file-type--anders"
          >
            Anders
          </span>
          <span
            className="resources-item__description-text"
          >
            http://acc.ergens
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="resources-item__file-type resources-item__file-type--anders"
            >
              Anders
            </span>,
            <span
              className="resources-item__description-text"
            >
              http://acc.ergens
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Anders",
              "className": "resources-item__file-type resources-item__file-type--anders",
            },
            "ref": null,
            "rendered": "Anders",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "http://acc.ergens",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "http://acc.ergens",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__file-type resources-item__file-type--anders"
            >
              Anders
            </span>
            <span
              className="resources-item__description-text"
            >
              http://acc.ergens
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="resources-item__file-type resources-item__file-type--anders"
              >
                Anders
              </span>,
              <span
                className="resources-item__description-text"
              >
                http://acc.ergens
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Anders",
                "className": "resources-item__file-type resources-item__file-type--anders",
              },
              "ref": null,
              "rendered": "Anders",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "http://acc.ergens",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "http://acc.ergens",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`The ResourcesItem component renders with title, description and date: leaving out file size for websites and APIs 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "ams:distributionType": "web",
        "ams:purl": "http://acc.ergens",
        "dcat:accessURL": "http://ergens",
        "dcat:byteSize": "0",
        "dcat:mediaType": "n/a",
        "dct:description": "omschrijving",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {},
      }
    }
    schemaProps={
      Object {
        "ams:distributionType": Object {
          "enum": Array [
            "api",
            "file",
            "web",
          ],
          "enumNames": Array [
            "API/Service",
            "Bestand",
            "Website",
          ],
        },
        "dcat:mediaType": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__file-type resources-item__file-type--website"
          >
            Website
          </span>
          <span
            className="resources-item__description-text"
          >
            omschrijving
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="resources-item__file-type resources-item__file-type--website"
            >
              Website
            </span>,
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Website",
              "className": "resources-item__file-type resources-item__file-type--website",
            },
            "ref": null,
            "rendered": "Website",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "omschrijving",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "omschrijving",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__file-type resources-item__file-type--website"
            >
              Website
            </span>
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="resources-item__file-type resources-item__file-type--website"
              >
                Website
              </span>,
              <span
                className="resources-item__description-text"
              >
                omschrijving
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Website",
                "className": "resources-item__file-type resources-item__file-type--website",
              },
              "ref": null,
              "rendered": "Website",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "omschrijving",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "omschrijving",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`The ResourcesItem component renders with title, description, format, file size and date 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "ams:distributionType": "file",
        "ams:purl": "http://acc.ergens",
        "dcat:accessURL": "http://ergens",
        "dcat:byteSize": "666",
        "dcat:mediaType": "application/pdf",
        "dct:description": "omschrijving",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {
          "dct:modified": "2017-11-30",
        },
      }
    }
    schemaProps={
      Object {
        "ams:distributionType": Object {
          "enum": Array [
            "api",
            "file",
            "web",
          ],
          "enumNames": Array [
            "API/Service",
            "Bestand",
            "Website",
          ],
        },
        "dcat:mediaType": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          666 B
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__file-type resources-item__file-type--pdf"
          >
            PDF
          </span>
          <span
            className="resources-item__description-text"
          >
            omschrijving
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "666 B",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "666 B",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="resources-item__file-type resources-item__file-type--pdf"
            >
              PDF
            </span>,
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PDF",
              "className": "resources-item__file-type resources-item__file-type--pdf",
            },
            "ref": null,
            "rendered": "PDF",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "omschrijving",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "omschrijving",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            666 B
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__file-type resources-item__file-type--pdf"
            >
              PDF
            </span>
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "666 B",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "666 B",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="resources-item__file-type resources-item__file-type--pdf"
              >
                PDF
              </span>,
              <span
                className="resources-item__description-text"
              >
                omschrijving
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PDF",
                "className": "resources-item__file-type resources-item__file-type--pdf",
              },
              "ref": null,
              "rendered": "PDF",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "omschrijving",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "omschrijving",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
